### DISCUSSION TITLE
add and close

### DISCUSSION THREAD

**Discussion by nenge123** [gildas-lormeau/zip.js#338] [Ideas]
if zipwrite#add not have await

then zipwrite#close will null file

how about using Promise.all check in when i close?

**Comment by gildas-lormeau** [comment]
Thank you for the suggestion, I agree it's a good idea so it's implemented in the version 2.4.26 that I just published.

**Reply by nenge123** [reply]
hey

**Comment by nenge123** [comment]
zip#Data64URIReader
is too long
how about add help func?
```
zip#W = (method,arr)=>zip#R(method,arr,'Writer');
zip#R = (method,arr,mode)=>{
method= method.tolowcase();
switch(method){
   case 'blob':
   case 'file':
   method = 'Blob';
break;
}
arr = arr||[];
if(arr in inseof  String)arr=[arr];
mode = mode||'Reader';
return new zip[method+mode].apply(zip,arr);

}
```

**Reply by gildas-lormeau** [reply]
You can use `zip.FS` if you want such helpers see https://github.com/gildas-lormeau/zip.js/blob/master/tests/all/test-fs.js for example.

**Reply by nenge123** [reply]
okay
